pkg/cmd/flags/logger.go

```diff
@@ -3,42 +3,100 @@ package flags
 import (
 	"fmt"
 	"io"
+	"strings"
+	"time"
+	"unicode"
 
 	"github.com/aquasecurity/tracee/pkg/logger"
 )
 
 func logHelp() string {
-	return `Control logger level priority.
+	return `Control logger options - aggregation and level priority.
 
 Possible options:
-  --log debug               | debug log level
-  --log info                | information log level (default) 
-  --log warn                | warning log level
-  --log error               | error log level
-  --log panic               | panic log level
+  --log aggregate                | turns log aggregation on, delaying output (default is off)
+  --log aggregate:interval       | turns log aggregation on, delaying output to every 'interval' (s, m)
+  --log debug                    | debug log level
+  --log info                     | information log level
+  --log warn                     | warning log level
+  --log error                    | error log level (default)
+  --log panic                    | panic log level
+
+Examples:
+  --log debug                    | outputs debug level logs
+  --log debug --log aggregate    | outputs aggregated debug level logs every 3 seconds (default)
+  --log aggregate:5s             | outputs aggregated logs every 5 seconds
 `
 }
 
-func PrepareLogger(logLevel string, w io.Writer) error {
-	var lvl logger.Level
-
-	switch logLevel {
-	case "debug":
-		lvl = logger.DebugLevel
-	case "info":
-		lvl = logger.InfoLevel
-	case "warn":
-		lvl = logger.WarnLevel
-	case "error":
-		lvl = logger.ErrorLevel
-	case "fatal":
-		lvl = logger.FatalLevel
-	default:
-		return fmt.Errorf("invalid log level: %s, use '--log help' for more info", logLevel)
+func InvalidLogOption(opt string) error {
+	return fmt.Errorf("invalid log option: %s, use '--log help' for more info", opt)
+}
+
+func PrepareLogger(logOptions []string, w io.Writer) (*logger.LoggerConfig, error) {
+	var (
+		agg      bool
+		interval = logger.DefaultFlushInterval
+		lvl      = logger.DefaultLevel
+		err      error
+	)
+
+	for _, opt := range logOptions {
+		// parse aggregate option
+		agg = strings.HasPrefix(opt, "aggregate")
+		if agg && !strings.HasSuffix(opt, "aggregate") {
+			vals := strings.Split(opt, ":")
+			if len(vals) != 2 || len(vals[1]) <= 1 {
+				return nil, InvalidLogOption(opt)
+			}
+
+			// handle only seconds and minutes
+			timeSuffix := vals[1][len(vals[1])-1:][0]
+			if timeSuffix != 's' && timeSuffix != 'm' {
+				return nil, InvalidLogOption(opt)
+			}
+			prevByte := vals[1][len(vals[1])-2:][0]
+			if timeSuffix == 's' && !unicode.IsDigit(rune(prevByte)) {
+				return nil, InvalidLogOption(opt)
+			}
+
+			// parse interval
+			interval, err = time.ParseDuration(vals[1])
+			if err != nil {
+				return nil, InvalidLogOption(opt)
+			}
+		}
+		if agg {
+			continue
+		}
```

Comment:
I reduced scope. https://github.com/aquasecurity/tracee/compare/b1a7b9a2cb6a1d537f8c732e6664be56af0f9821..63c8c198316bc711539c37b537604f9b0849eb8d#diff-b3ea339a5992ffc7674c4b4a72b6d1a45332a11e5b04ce655d369d960555345aR45-R69

Reply:
Introduced a bug and reverted.